### PR TITLE
Support for adding custom blocks in levelbuilder

### PR DESCRIPTION
**Pull request by balderdash**
Add some json defined blocks:
image

Then they show up in the toolbox:
image

### CODE REVIEW

File: dashboard/app/views/levels/editors/_gamelab_jr.html.haml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      Here is an example.
+    %pre
+      :preserve
+        {
+          "func": "isDestroyed",
+          "args": [
+            { "name": "SPRITE" }
+          ],
+          "blockText": "{SPRITE} is destroyed?",
+          "returnType": "Boolean",
```
Comment (joshlory) [inline dashboard/app/views/levels/editors/_gamelab_jr.html.haml:17]: Extra comma here is making the example JSON un-parsable.

File: dashboard/app/views/levels/editors/_gamelab_jr.html.haml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  = f.label :custom_blocks, 'Custom block defintions'
+  %p
+    Add some custom JSON block definitions
+    <a href="#" onClick="toggleBlock('custom_blocks_more'); return false">Read more...</a>
+  #custom_blocks_more{style: 'display: none'}
+    %p
+      Here is an example.
+    %pre
+      :preserve
+        {
```
Comment (joshlory) [inline dashboard/app/views/levels/editors/_gamelab_jr.html.haml:11]: It looks like the field is expecting an array of objects.

**Comment by joshlory** [comment]
Tagging @mrjoshida so he's aware this is coming down the pipeline.

File: dashboard/app/views/levels/editors/_gamelab_jr.html.haml (hunk 1 above)
Comment (balderdash) [reply dashboard/app/views/levels/editors/_gamelab_jr.html.haml:17]: Added [jsonic](https://www.npmjs.com/package/jsonic) parsing to make it less annoying to hand-craft the JSON here. If you don't quote keys or have trailing commas, the editor will automatically fix it up for you when you unfocus the textbox.

File: dashboard/app/views/levels/editors/_gamelab_jr.html.haml (hunk 2 above)
Comment (balderdash) [reply dashboard/app/views/levels/editors/_gamelab_jr.html.haml:11]: I also made it automatically wrap an object in an array if you have one block that you forgot to put into an array.

**Comment by mrjoshida** [comment]
This is insane!!!

**Pull request merged**